src/pages/home/report/ReportActionItem.js

```diff
@@ -70,59 +67,45 @@ const propTypes = {
 
 const defaultProps = {
     draftMessage: '',
-    hasOutstandingIOU: false,
 };
 
-class ReportActionItem extends Component {
-    constructor(props) {
-        super(props);
-        this.popoverAnchor = undefined;
-        this.state = {
-            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(props.action.reportActionID),
-        };
-        this.checkIfContextMenuActive = this.checkIfContextMenuActive.bind(this);
-        this.showPopover = this.showPopover.bind(this);
-        this.renderItemContent = this.renderItemContent.bind(this);
-        this.toggleReaction = this.toggleReaction.bind(this);
-    }
+function ReportActionItem(props) {
+    const [isContextMenuActive, setIsContextMenuActive] = React.useState(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    const previousDraftMessage = React.useRef('');
+    const textInput = React.useRef();
+    const popoverAnchor = React.useRef();
 
-    shouldComponentUpdate(nextProps, nextState) {
-        return this.props.displayAsGroup !== nextProps.displayAsGroup
-            || this.props.draftMessage !== nextProps.draftMessage
-            || this.props.isMostRecentIOUReportAction !== nextProps.isMostRecentIOUReportAction
-            || this.props.hasOutstandingIOU !== nextProps.hasOutstandingIOU
-            || this.props.shouldDisplayNewMarker !== nextProps.shouldDisplayNewMarker
-            || !_.isEqual(this.props.action, nextProps.action)
-            || this.state.isContextMenuActive !== nextState.isContextMenuActive;
-    }
+    React.useEffect(() => {
+        previousDraftMessage.current = props.draftMessage;
+    }, [props.draftMessage]);
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.draftMessage || !this.props.draftMessage) {
+    React.useEffect(() => {
+        if (previousDraftMessage.current || !props.draftMessage) {
             return;
         }
 
         // Only focus the input when user edits a message, skip it for existing drafts being edited of the report.
         // There is an animation when the comment is hidden and the edit form is shown, and there can be bugs on different mobile platforms
         // if the input is given focus in the middle of that animation which can prevent the keyboard from opening.
-        focusTextInputAfterAnimation(this.textInput, 100);
-    }
+        focusTextInputAfterAnimation(textInput.current, 100);
+    }, [props.draftMessage]);
 
-    checkIfContextMenuActive() {
-        this.setState({isContextMenuActive: ReportActionContextMenu.isActiveReportAction(this.props.action.reportActionID)});
-    }
+    const checkIfContextMenuActive = () => {
+        setIsContextMenuActive(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    };
```

Comment:
NAB, I don't get the name of this one. `checkIfContextMenuActive`.. it looks like we are not "checking" anything and actually setting something to state that can be computed from props... (at least in this case) 🤔

Reply:
oh yeah that's a great point, I honestly feel like we don't even need this function and we can just call the code inside directly.

Reply:
EDIT: renamed it to `toggleContextMenuFromActiveReportAction`